trajopt_optimizers/trajopt_optimizers_test_suite/src/joint_velocity_optimization_unit.cpp

```diff
@@ -0,0 +1,55 @@
+#include <trajopt_utils/macros.h>
+TRAJOPT_IGNORE_WARNINGS_PUSH
+#include <gtest/gtest.h>
+#include <iostream>
+
+#include <ifopt/problem.h>
+#include <ifopt/ipopt_solver.h>
+#include <console_bridge/console.h>
+TRAJOPT_IGNORE_WARNINGS_POP
+
+#include <trajopt_sqp/trust_region_sqp_solver.h>
+#include <trajopt_sqp/osqp_eigen_solver.h>
+#include <trajopt_ifopt/constraints/joint_position_constraint.h>
+#include <trajopt_ifopt/constraints/joint_velocity_constraint.h>
+#include <trajopt_ifopt/variable_sets/joint_position_variable.h>
+#include <trajopt_ifopt/costs/squared_cost.h>
+
+#include <joint_velocity_optimization_unit.hpp>
+
+/**
+@brief Joint position constraints with a squared velocity cost in between. Optimized using ipopt
+ */
+TEST_F(VelocityConstraintOptimization, velocity_constraint_optimization_ipopt)
+{
+  ifopt::Problem nlp_opt(nlp_);
+  ifopt::IpoptSolver solver;
+  solver.SetOption("derivative_test", "first-order");
+  solver.SetOption("linear_solver", "mumps");
+  //  ipopt.SetOption("jacobian_approximation", "finite-difference-values");
+  solver.SetOption("jacobian_approximation", "exact");
+  solver.SetOption("print_level", 5);
+
+  runTests<ifopt::IpoptSolver>(solver, nlp_opt);
+}
+
+
+/**
+ @brief Joint position constraints with a squared velocity cost in between. Optimized using trajopt_sqp
+ */
+TEST_F(VelocityConstraintOptimization, velocity_constraint_optimization_trajopt_sqp)
+{
+  ifopt::Problem nlp_trajopt_sqp(nlp_);
+  auto qp_solver = std::make_shared<trajopt_sqp::OSQPEigenSolver>();
+  trajopt_sqp::TrustRegionSQPSolver solver(qp_solver);
+  qp_solver->solver_.settings()->setVerbosity(DEBUG);
+  qp_solver->solver_.settings()->setWarmStart(true);
+  qp_solver->solver_.settings()->setAbsoluteTolerance(1e-4);
+  qp_solver->solver_.settings()->setRelativeTolerance(1e-6);
+  qp_solver->solver_.settings()->setMaxIteraction(8192);
+  qp_solver->solver_.settings()->setPolish(true);
+  qp_solver->solver_.settings()->setAdaptiveRho(false);
+
+  solver.verbose = DEBUG;
+  runTests<trajopt_sqp::TrustRegionSQPSolver>(solver, nlp_trajopt_sqp);
```

Comment:
I think we want to move these into the package where they are implemented. So this one will go in trajopt_sqp/test.

Reply:
So should the src directory just be renamed to the test directory?

Reply:
Effectively yes. All of this code should go in test. See how we do it on the other trajopt packages.